### PR TITLE
fix: drive pipeline forever in debug.continuous

### LINKED ISSUE #2044: hive `ethereum/sync` test fails due to broken `debug.continuous`
### Describe the bug

The test used to work, but now fails. The `continuous` downloader does not start, and it might be related to recent changes in Consensus.

### Steps to reproduce

Either run `--debug.continuous`:
```
reth node --debug.continuous
```

Or run [hive](#851):
```
./hive --client reth --sim ethereum/sync
```

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by Rjected**
Fixes #2044

The `--debug.continuous` downloader broke because it relied on the pipeline to run forever. The pipeline was previously spawned directly by the `reth node` command in a loop, but is now driven by the `BeaconConsensusEngine`. The `BeaconConsensusEngine` will not run the pipeline again if there is no new pipeline target.

This adds a flag to the `BeaconConsensusEngine` that, if set, will prompt the pipeline to run again, even if there is no new pipeline target. Because `HeaderSyncMode::Continuous` causes the downloader to always have a new / higher tip, this restores the continuous download functionality.

Sends the genesis hash as a fork choice update to the `BeaconConsensusEngine` to kick off download if `--debug.continuous` is set.

### CODE REVIEW

**Review by rkrasiuk — COMMENTED** [review]
lgtm, cleaned up a bit and fixed run metric for continuous

**Review by rkrasiuk — APPROVED** [review]

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -212,12 +212,25 @@ impl Command {
         info!(target: "reth::cli", peer_id = %network.peer_id(), local_addr = %network.local_addr(), "Connected to P2P network");
         debug!(target: "reth::cli", peer_id = ?network.peer_id(), "Full peer ID");
 
+        let (consensus_engine_tx, consensus_engine_rx) = unbounded_channel();
+
+        // Forward genesis as forkchoice state to the consensus engine.
+        // This will allow the downloader to start
         if self.debug.continuous {
             info!(target: "reth::cli", "Continuous sync mode enabled");
+            let (tip_tx, _tip_rx) = oneshot::channel();
```
Comment (rkrasiuk) [inline bin/reth/src/node/mod.rs:221]: on a second thought, wouldn't this cause an error if it's dropped?

Comment (rkrasiuk) [reply bin/reth/src/node/mod.rs:221]: i guess not, because we ignore closed receiver errors in the engine

**Comment by rkrasiuk** [comment]
would also be nice to make `--debug.continuous` conflict with `--debug.tip`

**Review by gakonst — APPROVED** [review]
@Rjected said he tested on Hive locally so taking his word here - LGTM otherwise

**Pull request merged**